presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -289,19 +288,10 @@ private void alterTable(HiveIdentity identity, String databaseName, String table
             return ImmutableMap.of();
         }
 
-        Function<org.apache.hadoop.hive.metastore.api.Partition, Partition> fromMetastoreApiPartition = ThriftMetastoreUtil::fromMetastoreApiPartition;
-        boolean isAvroTableWithSchemaSet = delegate.getTable(identity, databaseName, tableName)
-                .map(ThriftMetastoreUtil::isAvroTableWithSchemaSet)
-                .orElse(false);
-        if (isAvroTableWithSchemaSet) {
-            List<FieldSchema> schema = delegate.getFields(identity, databaseName, tableName).get();
-            fromMetastoreApiPartition = partition -> fromMetastoreApiPartition(partition, schema);
```

Comment:
This leads to inconsistency.
For table schema, we're considering Avro schema file (`io.prestosql.plugin.hive.metastore.thrift.BridgingHiveMetastore#getTable`) (which is correct, otherwise `DESC avro_table` would not be correct).

However, for fetching partition schema we no longer do so. This may lead to us thinking there is table/partition schema mismatch (eg that cannot be handled within permissible schema evolution), leading to query failure or misbehavior. 

If my reasoning is unsound, please add a code comment here explaining why we don't need to consider actual schema of a partition.

Reply:
> For table schema, we're considering Avro schema file
> However, for fetching partition schema we no longer do so. 

We still do, because we replace the partition schema on access via `getPartitionColumns()` with the schema of the `Table` object, which comes from the Avro schema file via `getTable()`.

Reply:
My understanding is that for Avro, both the table and partition schemas are garbage. The actual schema comes from the Thrift `get_fields()` call.

(this leads to the obvious question of why the metastore doesn't do this substitution itself when returning the storage descriptor for `Table` and `Partition` objects)